numba/extending.py

```diff
@@ -166,3 +169,118 @@ def struct_getattr_impl(context, builder, typ, val):
         attrty = get_attr_fe_type(typ)
         attrval = getattr(val, struct_attr)
         return impl_ret_borrowed(context, builder, attrty, attrval)
+
+
+class _LLVMCall(object):
+    """
+    Dummy callable for llvm_call
+    """
+    _memo = weakref.WeakValueDictionary()
+    __uuid = None
+
+    def __init__(self, name, defn):
+        self._name = name
+        self._defn = defn
+
+    @property
+    def _uuid(self):
+        """
+        An instance-specific UUID, to avoid multiple deserializations of
+        a given instance.
+
+        Note this is lazily-generated, for performance reasons.
+        """
+        u = self.__uuid
+        if u is None:
+            u = str(uuid.uuid1())
+            self._set_uuid(u)
+        return u
+
+    def _set_uuid(self, u):
+        assert self.__uuid is None
+        self.__uuid = u
+        self._memo[u] = self
+
+    def _register(self):
+        from .typing.templates import make_llvm_call_template, infer_global
+
+        template = make_llvm_call_template(self, self._defn, self._name)
+        infer(template)
+        infer_global(self, types.Function(template))
+
+    def __call__(self, *args, **kwargs):
+        """
+        This is only defined to pretend to be a callable from CPython.
+        """
+        msg = '{0} is not usable in pure-python'.format(self)
+        raise NotImplementedError(msg)
+
+    def __repr__(self):
+        return "<llvm_call {0}>".format(self._name)
+
+    def __reduce__(self):
```

Comment:
An LLVMCall object could be referenced by a jit function as a global.   When the jit function is serialized, its global references are also serialized.